extensions/clickup/src/capture.tsx

```diff
@@ -0,0 +1,73 @@
+import { Form, ActionPanel, Action, showToast, Toast, popToRoot } from "@raycast/api";
+import { useState } from "react";
+import { ClickUpClient } from "./utils/clickUpClient";
+import { TaskItem } from "./types/tasks.dt";
+import preferences from "./utils/preferences";
+
+interface FormValues {
+  name: string;
+  description: string;
+  dueDate: string;
+  priority: number;
+}
+
+export default function QuickCapture() {
+  // Error message for the primary TextArea
+  const [textErrorMessage, setTextErrorMessage] = useState<string | undefined>();
+  const [task, setTask] = useState<TaskItem | undefined>(undefined);
+
+  // Clear the text error message
+  const clearTextErrorMessage = () => setTextErrorMessage(undefined);
+
+  const handleSubmit = async (formValues: FormValues) => {
+    if (!formValues.name) return setTextErrorMessage("Required");
+
+    const data: Record<string, any> = { name: formValues.name };
+    if (formValues.description) {
+      data["description"] = formValues.description;
+    }
+    if (formValues.dueDate) {
+      data["due_date"] = new Date(formValues.dueDate).getTime();
+      console.log(data["due_date"]);
+    }
+    if (formValues.priority) {
+      data["priority"] = formValues.priority;
+    }
+
+    await ClickUpClient<TaskItem>(`/list/${preferences.listId}/task`, "POST", data)
```

Comment:
We can add a guard like
```       
if (!preferences.listId) return setTextErrorMessage("List Id is required");
```

but it won't be reachable code since `listId` is set in the manifest as a required preference field. If not set you'll see

![Screenshot 2023-01-12 at 10 07 17](https://user-images.githubusercontent.com/1596227/212145449-0b0877a5-29e1-4d3f-bac9-17ff3597ec50.png)

Until you add your default list.